lib/internal/http2.js

```diff
@@ -791,8 +785,7 @@ class Http2Outgoing extends Writable {
       chunk = Buffer.from(chunk, encoding);
     if (this.stream) {
       this[kBeginSend]();
-      this.bufferedCallback = callback;
-      this.stream.write(chunk, encoding, outWriteResume);
+      this.stream.write(chunk, encoding, callback);
```

Comment:
you should be able to get a boost here by doing:

```js
const res = this.stream.write(chunk, encoding)
if (!res) {
  this.stream.once('drain', cb)
} else {
  cb() // or process.nextTick(cb)
}
```

In this way we are doing as many writes we can before pausing for I/O to happen.